### ISSUE TITLE
Feature request: Quality-optimized subsampling

### ISSUE THREAD

**Issue by tdaede** [sekrit-twc/zimg#168]
There are methods for producing 4:2:0 subsampled video that aim to provide more optimal results once displayed (both subjectively, and via cross-coefficient PSNR and CIEDE2000 metrics):

https://norkin.org/pdf/SPIE_2016_HDR_conversion_metrics.pdf

Despite the HDR title, the methods are also useful for SDR content. The paper provides 3 methods, of which the "Closed Form 1" method is probably most suitable for a SIMD implementation.

Also, these algorithms currently only perturb the luma coefficient. This helps make the problem tractable but there are likely better algorithms that could potentially generate better chroma values as well.

HDRTools also has an open source implementation that can be checked against (HDRTools is more a reference implementation rather than a practical library like zimg).

**Comment by ValZapod** [comment]
Why do you think it is better than Lansoz 3 used by default by zscale?

**Comment by tdaede** [comment]
Yes, this is specifically for encoding only (or slightly more generally, when downsampling chroma).

The method in the paper converts RGB->YUV and does the downsample in one step. This doesn't map well to any of the existing zimg filter types. A YUV->YUV filter type can also be derived fairly efficiently as long as the colorspace is known, and the input is non constant luminance. I don't know if that would be worth implementing over simply forcing a conversion to RGB with an existing colorspace filter, though.

**Comment by ValZapod** [comment]
I believe this is what is meant 

> I have also seen the 1000 nit version on the X300 and X310. We ran the montage through Transkoder first, which fixes a lot of the YCbCr issues you mention, so the disc looks very much like the master in RGB. ICtCp would improve what you hare discussing

. here https://www.avsforum.com/threads/spears-munsil-uhd-hdr-benchmark-disc-discussion.3075780/post-58642776

I am not sure this is worse implementing though, since we slowely going to IPTPQc2 in Dolby Vision of profile 5 and ICtCp of ITU. Constant luminance was always an idea fix of people since 1989, so even back when BT.709 was being derived.

I suppose you should be aware that ITU-T H series supplement 18 described how chroma should be downsampled... And it is not this. I also see no chroma siting info at all there.

**Comment by ValZapod** [comment]
Why all the downvotes? You did not like propriatery SW mention?

**Comment by Artoria2e5** [comment]
The webp people has a different, iterative thing that touches not just luma. It's available as a a libyuvsharp your program can link to, but it's only intended for static images given the speed.

https://github.com/mozilla/mozjpeg/issues/193 is related
